src/Widgets/TerminalWidget.vala

```diff
@@ -547,12 +549,21 @@ namespace Terminal {
                 _("Are you sure you want to reset the terminal?"),
                 _("Reset"))
             ) {
+                // We need to clear any pending input else it can get unexpectedly executed
+                clear_pending_input ();
                 // This also clears the screen and the scrollback
                 // We know there is no foreground process so we can just feed the command in
-                feed_child ("reset\n".data);
+                feed_child ("\rreset\n".data);
             }
         }
 
+        public void clear_pending_input () {
+            // This is hacky but no obvious way to feed in escape sequences to clear the line
+            // Assume any pending input is less than 1000 chars.
+            string backspaces = string.nfill (1000, '\b');
+            feed_child (backspaces.data);
+        }
```

Comment:
Can't you just pass Ctrl+C signal to clear current line instead of this?

Reply:
Finally worked out how to do this with a signal as you suggested!  Thanks.